allennlp/data/vocabulary.py

```diff
@@ -112,7 +113,17 @@ def _read_pretrained_tokens(embeddings_file_uri: str) -> Set[str]:
     return tokens
 
 
-class Vocabulary:
+class RegistrableVocabulary(Registrable):
+    default_implementation = "vocabulary"
+
+    @classmethod
+    def from_params(cls, params: Params, instances: Iterable['adi.Instance'] = None):
```

Comment:
To make this backward compatible (and fix all the failing tests), you'll need to check if `params` has a type key and if not, use the default implementation of the vocabulary.